test_marketplace_login_from_app_details_page.py is failing on CI

This test is failing, as can be seen at [1].

Stack trace:

```
Traceback (most recent call last):
  File "/var/jenkins/1/workspace/flame-kk-319.mozilla-b2g32_v2_0.nightly.ui.marketplace-tests/.env/local/lib/python2.7/site-packages/marionette/marionette_test.py", line 170, in run
    testMethod()
  File "/var/jenkins/1/workspace/flame-kk-319.mozilla-b2g32_v2_0.nightly.ui.marketplace-tests/marketplacetests/tests/test_marketplace_login_from_app_details_page.py", line 40, in test_marketplace_login_from_app_details_page
    self.assertEqual(marketplace.notification_message, "Your review was successfully posted. Thanks!")
TEST-UNEXPECTED-FAIL | test_marketplace_login_from_app_details_page.py test_marketplace_login_from_app_details_page.TestMarketplaceLoginFromAppDetailsPage.test_marketplace_login_from_app_details_page | AssertionError: u'You have been signed in' != 'Your review was successfully posted. Thanks!'
```

I am going to look into this this morning.

[1] https://webqa-ci.mozilla.com/job/flame-kk-319.mozilla-b2g32_v2_0.nightly.ui.marketplace-tests/117/HTML_Report/

--- a/marketplacetests/marketplace/app.py
+++ b/marketplacetests/marketplace/app.py
@@ -4,6 +4,7 @@
 
 from marionette.by import By
 from marionette.keys import Keys
+from marionette.wait import Wait
 from gaiatest.apps.base import Base
 
 
@@ -72,7 +73,10 @@ class Marketplace(Base):
         self.wait_for_element_displayed(*self._offline_message_locator)
         return self.marionette.find_element(*self._offline_message_locator).text
 
-    def wait_for_notification_message_displayed(self):
+    def wait_for_notification_message_displayed(self, message=None):
+        if message:
+            Wait(marionette=self.marionette).until(
+                lambda m: self.notification_message == message)
         self.wait_for_element_displayed(*self._notification_locator)
 
     def wait_for_notification_message_not_displayed(self):

--- a/marketplacetests/tests/test_marketplace_login_from_app_details_page.py
+++ b/marketplacetests/tests/test_marketplace_login_from_app_details_page.py
@@ -34,9 +34,8 @@ class TestMarketplaceLoginFromAppDetailsPage(MarketplaceGaiaTestCase):
         review_box = AddReview(self.marionette)
 
         review_box.write_a_review(rating, body)
-        marketplace.wait_for_notification_message_displayed()
+        marketplace.wait_for_notification_message_displayed('Your review was successfully posted. Thanks!')
 
         # Check if review was added correctly
-        self.assertEqual(marketplace.notification_message, "Your review was successfully posted. Thanks!")
         self.assertEqual(details_page.first_review_rating, rating)
         self.assertEqual(details_page.first_review_body, body)

